File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/update_experiment_tags_details.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class UpdateExperimentTagsDetails(object):
    """
    Details of the Experiment tags to update.
    """

    def __init__(self, **kwargs):
        """
        Initializes a new UpdateExperimentTagsDetails object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param experiment_id:
            The value to assign to the experiment_id property of this UpdateExperimentTagsDetails.
        :type experiment_id: str

        :param set_tags:
            The value to assign to the set_tags property of this UpdateExperimentTagsDetails.
        :type set_tags: list[oci.aidataplatform_dp.models.ExperimentTag]

        :param delete_tags:
            The value to assign to the delete_tags property of this UpdateExperimentTagsDetails.
        :type delete_tags: list[oci.aidataplatform_dp.models.ExperimentTagKey]

        """
        self.swagger_types = {
            'experiment_id': 'str',
            'set_tags': 'list[ExperimentTag]',
            'delete_tags': 'list[ExperimentTagKey]'
        }

        self.attribute_map = {
            'experiment_id': 'experiment_id',
            'set_tags': 'set_tags',
            'delete_tags': 'delete_tags'
        }

        self._experiment_id = None
        self._set_tags = None
        self._delete_tags = None

    @property
    def experiment_id(self):
        """
        **[Required]** Gets the experiment_id of this UpdateExperimentTagsDetails.
        Unique identifier for the experiment.


        :return: The experiment_id of this UpdateExperimentTagsDetails.
        :rtype: str
        """
        return self._experiment_id

    @experiment_id.setter
    def experiment_id(self, experiment_id):
        """
        Sets the experiment_id of this UpdateExperimentTagsDetails.
        Unique identifier for the experiment.


        :param experiment_id: The experiment_id of this UpdateExperimentTagsDetails.
        :type: str
        """
        self._experiment_id = experiment_id

    @property
    def set_tags(self):
        """
        Gets the set_tags of this UpdateExperimentTagsDetails.
        Experiment tags to set


        :return: The set_tags of this UpdateExperimentTagsDetails.
        :rtype: list[oci.aidataplatform_dp.models.ExperimentTag]
        """
        return self._set_tags

    @set_tags.setter
    def set_tags(self, set_tags):
        """
        Sets the set_tags of this UpdateExperimentTagsDetails.
        Experiment tags to set


        :param set_tags: The set_tags of this UpdateExperimentTagsDetails.
        :type: list[oci.aidataplatform_dp.models.ExperimentTag]
        """
        self._set_tags = set_tags

    @property
    def delete_tags(self):
        """
        Gets the delete_tags of this UpdateExperimentTagsDetails.
        Experiment tags to delete


        :return: The delete_tags of this UpdateExperimentTagsDetails.
        :rtype: list[oci.aidataplatform_dp.models.ExperimentTagKey]
        """
        return self._delete_tags

    @delete_tags.setter
    def delete_tags(self, delete_tags):
        """
        Sets the delete_tags of this UpdateExperimentTagsDetails.
        Experiment tags to delete


        :param delete_tags: The delete_tags of this UpdateExperimentTagsDetails.
        :type: list[oci.aidataplatform_dp.models.ExperimentTagKey]
        """
        self._delete_tags = delete_tags

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/UpdateExperimentTagsDetails.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;



/**
 * Details of the Experiment tags to update.
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=UpdateExperimentTagsDetails.Builder.class)

public final class UpdateExperimentTagsDetails  {
    @Deprecated
    @java.beans.ConstructorProperties({"experimentId", "setTags", "deleteTags"})
    public UpdateExperimentTagsDetails(String experimentId, java.util.List<ExperimentTag> setTags, java.util.List<ExperimentTagKey> deleteTags) {
        super();
        this.experimentId = experimentId;
        this.setTags = setTags;
        this.deleteTags = deleteTags;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * Unique identifier for the experiment.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("experiment_id")
private String experimentId;

        /**
         * Unique identifier for the experiment.
         * @param experimentId the value to set
         * @return this builder
         **/
        

public Builder experimentId(String experimentId) {
    this.experimentId = experimentId;
    return this;
}
            /**
     * Experiment tags to set
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("set_tags")
private java.util.List<ExperimentTag> setTags;

        /**
         * Experiment tags to set
         * @param setTags the value to set
         * @return this builder
         **/
        

public Builder setTags(java.util.List<ExperimentTag> setTags) {
    this.setTags = setTags;
    return this;
}
            /**
     * Experiment tags to delete
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("delete_tags")
private java.util.List<ExperimentTagKey> deleteTags;

        /**
         * Experiment tags to delete
         * @param deleteTags the value to set
         * @return this builder
         **/
        

public Builder deleteTags(java.util.List<ExperimentTagKey> deleteTags) {
    this.deleteTags = deleteTags;
    return this;
}


        public UpdateExperimentTagsDetails build() {
            UpdateExperimentTagsDetails model = new UpdateExperimentTagsDetails(this.experimentId
                , this.setTags
                , this.deleteTags);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(UpdateExperimentTagsDetails model) {
                this.experimentId(model.getExperimentId());
    this.setTags(model.getSetTags());
    this.deleteTags(model.getDeleteTags());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * Unique identifier for the experiment.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("experiment_id")
    private final String experimentId;

        /**
     * Unique identifier for the experiment.
     * @return the value
     **/
    
    public String getExperimentId() {
        return experimentId;
    }


        /**
     * Experiment tags to set
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("set_tags")
    private final java.util.List<ExperimentTag> setTags;

        /**
     * Experiment tags to set
     * @return the value
     **/
    
    public java.util.List<ExperimentTag> getSetTags() {
        return setTags;
    }


        /**
     * Experiment tags to delete
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("delete_tags")
    private final java.util.List<ExperimentTagKey> deleteTags;

        /**
     * Experiment tags to delete
     * @return the value
     **/
    
    public java.util.List<ExperimentTagKey> getDeleteTags() {
        return deleteTags;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("UpdateExperimentTagsDetails(");
        sb.append("experimentId=").append(String.valueOf(this.experimentId));
        sb.append(", setTags=").append(String.valueOf(this.setTags));
        sb.append(", deleteTags=").append(String.valueOf(this.deleteTags));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof UpdateExperimentTagsDetails)) {
            return false;
        }

        UpdateExperimentTagsDetails other = (UpdateExperimentTagsDetails) o;
        return java.util.Objects.equals(this.experimentId, other.experimentId) &&
            java.util.Objects.equals(this.setTags, other.setTags) &&
            java.util.Objects.equals(this.deleteTags, other.deleteTags);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.experimentId == null ? 43 : this.experimentId.hashCode());
        result = (result * PRIME) + (this.setTags == null ? 43 : this.setTags.hashCode());
        result = (result * PRIME) + (this.deleteTags == null ? 43 : this.deleteTags.hashCode());
        return result;
    }


}
